src/relay/backend/contrib/ethosn/ethosn_api.cc

```diff
@@ -213,27 +227,29 @@ EthosnError EthosnAPI::QnnFullyConnected(const Expr& expr, FullyConnectedParams*
                                       data_data_type, sl::DataFormat::NHWC, data_q_info);
 
   // Create weights info
-  const auto* weights_dtype = dense->args[1]->checked_type().as<TensorTypeNode>();
+  Constant weights_data = Downcast<Constant>(dense->args[1]);
+  weights_data = TransposeWeights(weights_data, "OI", "IO");
```

Comment:
nit: I am in favor of full 4 dim conversion just to be consistent with the Relay supported kernel layouts.

Reply:
I think the issue is that the input here is 2-d so `MakeTranspose` will expect a transpose array with only 2 values

Reply:
my bad!